### DISCUSSION TITLE
How do you pass environment variables to vcpkg_execute_required_process?

### DISCUSSION THREAD

**Discussion by Naville** [microsoft/vcpkg#41867] [Q&A]
Title.
I'm writing portfile for a project that uses custom buildsystem and requires CC / CXX / C*FLAGS to be passed in as env vars

**Comment by Osyotr** [comment]
Have you tried `set(ENV{CC} "...")`?

**Reply by Naville** [reply]
Doesn't seem to work for me. Though I havn't tried if it's due to the target build system didn't handle it properly or vcpkg didn't pass at all

**Reply by dg0yt** [reply]
How do you invoke the "custom buildsystem"? 

There is no general filtering of environment variables when starting a subprocess from `portfile.cmake`. But some maintainer functions (`vcpkg_configure_make`) do set these variables in a controlled way. (And this is useful.)

**Reply by Naville** [reply]
I assume this is correct from vcpkg side then:
```
vcpkg_execute_required_process(
    COMMAND
    "${PYTHON_WITH_SCONS}"
    "-m"
    "SCons"
    .....
    "-C"
    "${DEB_BUILD_DIR}"
    "all"
    "-j"
    "${PROCS}"
    WORKING_DIRECTORY
    "${SOURCE_PATH}"
    LOGNAME
    "build-${TARGET_TRIPLET}-dbg"
    SAVE_LOG_FILES
    ${TARGET_TRIPLET}-dbg.log)
```

I'll check if the custom buildsystem is broken internally :)

**Reply by dg0yt** [reply]
scons, I see. There is no good example of this in vcpkg, and indeed no ready-to-use maintainer function.

(For some non-standard projects, I use some configure/make polyfill to leverage the standard configure/make maintainer functions. Example: libcap)
